### ISSUE TITLE
Official Riverpod support

### ISSUE THREAD

**Issue by devberkay** [flame-engine/flame#2353]
# Problem to solve
Games need complex state management solutions and Riverpod is the core dependency for many flutter developers like me. 

# Proposal
It would be amazing if we could see a collabaration in the future as it was done with flame_bloc package.

# More information
@rrousselGit

**Comment by spydon** [comment]
Lucky for you there is already a package for this, which we soon will be reviewing and bringing in to the monorepo.
https://github.com/markvideon/flame_riverpod
https://pub.dev/packages/flame_riverpod

**Comment by spydon** [comment]
@markvideon do you know when you'll put up a PR for this? :)

**Comment by rrousselGit** [comment]
Making an official Riverpod implementation involves more work.

As is, it's not going to work with any tools like riverpod_lint & upcoming.

**Comment by spydon** [comment]
> Making an official Riverpod implementation involves more work.
> 
> As is, it's not going to work with any tools like riverpod_lint & upcoming. 

Well depends on what is meant with official here I guess, if he means an official bridge package from the Flame side (like flame_bloc) or official from the Riverpod side.

**Comment by rrousselGit** [comment]
I assume it should be official for both sides at the same time. Otherwise it's not truly official right? :)
Where the project sits doesn't really matter. I'm fine with it being in the flame repository.

But it should be compatible with riverpod_lint, which is a core Riverpod package.  Nothing blocking, but I'm sure it'd be inconvenient for Riverpod users of flame's consumer broke the various refactors & lints

**Comment by spydon** [comment]
> I assume it should be official for both sides at the same time. Otherwise it's not truly official right? :)

True true. :)

> But it should be compatible with riverpod_lint, which is a core Riverpod package. Nothing blocking, but I'm sure it'd be inconvenient for Riverpod users if flame's consumer broke the various refactors & lints

Alright, let's create an issue for that once the initial version is merged into the monorepo.
I ran the linter on the the example that was provided in the repository and there were no warnings reported in there, but you're probably right that refactors wouldn't work like they should.

**Comment by markvideon** [comment]
> @markvideon do you know when you'll put up a PR for this? :)

I'll try to get to it this week

**Comment by markvideon** [comment]
@spydon can probably close this one?
